components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -598,14 +606,54 @@ void RewardsServiceImpl::CreateRewardsWallet(
         observer.OnRewardsWalletUpdated();
       }
 
-      std::move(callback).Run(ledger::mojom::Result::LEDGER_OK);
+      std::move(callback).Run(result);
     };
 
     self->bat_ledger_->CreateRewardsWallet(
-        base::BindOnce(on_created, self, std::move(callback)));
+        country, base::BindOnce(on_created, self, std::move(country),
```

Comment:
The order in which function arguments get evaluated is unspecified, therefore if `base::BindOnce<>()` runs first, you're out of luck, as `country` gets passed to `CreateRewardsWallet()` as a moved-from object.

Reply:
Of course - thanks for catching!

Reply:
@zenparsing Actually, this was the reason for calling `POST /v3/wallet/brave` instead of `POST /v4/wallets` (mentioned [here](https://github.com/brave/brave-core/pull/15199#issuecomment-1262511848)). Now it's working fine. 👍🏻